### PR TITLE
Flickr Widget: build Feed URLs manually based on user or group ID

### PR DESCRIPTION
**Pull request by jeherve**
Building the feed ourselves avoids issues when a user enters the wrong RSS format, or URL format.

#### Testing instructions:

With that PR, one can now use the following feed formats, that will all be converted into a compatible Flickr RSS feed:

- https://www.flickr.com/services/feeds/photos_public.gne?id=34430197@N04&lang=en-us&format=rss
- https://api.flickr.com/services/feeds/groups_pool.gne?id=417611@N24&format=rss2
- https://api.flickr.com/services/feeds/groups_pool.gne?id=417611@N24&format=rss_200_enc
- https://api.flickr.com/services/feeds/groups_pool.gne?id=417611@N24
- https://api.flickr.com/services/feeds/groups_pool.gne?id=417611@N24&format=atom
- https://api.flickr.com/services/feeds/groups_pool.gne?id=417611@N24&format=rdf
- https://api.flickr.com/services/feeds/photos_public.gne?id=11023327@N08&lang=fr-fr&format=rss_200

Reported here:
p8oabR-2x-p2

### CODE REVIEW

File: modules/widgets/flickr.php [hunk 1]
```diff
@@ -71,8 +71,34 @@ public function widget( $args, $instance ) {
...
 				: htmlspecialchars_decode( $instance['flickr_rss_url'] );
 
-			// We want to use the HTTPS version so the URLs in the API response are also HTTPS and we avoid mixed-content warnings.
-			$rss_url = preg_replace( '!^http://api.flickr.com/!i', 'https://api.flickr.com/', $rss_url );
+			/**
+			 * Parse the URL, and rebuild a URL that's sure to display images.
+			 * Some Flickr Feeds do not display images by default.
+			 */
+			$flickr_parameters = parse_url( $rss_url );
+			parse_str( $flickr_parameters['query'], $vars );
```
Comment (dereksmart) [inline modules/widgets/flickr.php]: Might be a good idea to be more defensive here.  It throws errors on the front end if you use a garbage url like `whatever.com`

Comment (jeherve) [inline modules/widgets/flickr.php]: @dereksmart Would something like this work for you?
75f18ad19415b1424a5f3d3463d26b20de9328de

We don't accept random URLs anymore.

**Comment by dereksmart** [comment]
@jeherve yeah this works for me.  I pushed a very small commit that just fixes the [inline comment](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#5-1-single-line-comments) syntaxes, but this is good to 🐑

**Comment by dereksmart** [comment]
The only thing I'd suggest is adding a note similar to what we do for no photos found, if the URL is not a valid flickr feed.  As it is with this PR now, the URL could be `garbageman.comasdjklfh`, and we will still show random images with no message about the URL being wrong.  

Only add if you feel it's worth it.

**Comment by jeherve** [comment]
@dereksmart Yep, I agree that would be nice. I think it's okay to ship the widget as it is, though, and add more notes directly to the widget settings as suggested in #7096, in a v2.

**Comment by eliorivero** [comment]
Cherry picked to 4.9 f08a24c41e9a76dde48b4a6b2e80fd660bda7082

**Pull request merged**